packages/workspace/src/workspace/hidden_values.ts

```
@@ -383,14 +382,61 @@ const isHiddenField = async (
       || isHiddenField(baseType, fieldPath.slice(0, -1), hiddenValue, elementsSource)
 }
 
+const diffForHidden = (lhs: unknown, rhs: unknown): unknown => {
+  if (lhs === rhs) {
+    return undefined
+  }
+  if (rhs === undefined) {
+    return lhs
+  }
+
+  if (Array.isArray(lhs)) {
+    if (!Array.isArray(rhs) || lhs.length !== rhs.length) {
+      // should never happen
+      log.warn('Unexpected diff found in array: %s', safeJsonStringify([lhs, rhs]))
+      return lhs
+    }
+    const arrayDiff = lhs.map((val, idx) => diffForHidden(val, rhs[idx]))
+    if (arrayDiff.some(val => !_.isEmpty(val))) {
+      return arrayDiff
+    }
+    return undefined
```

Since we don't really support hidden values within arrays (either the whole array is hidden and it would have been caught in line 389, or non of it is hidden) I think the code here should always return `undefined`.
We can keep the checks and validations, but I think both cases where the arrays don't match should log an error and return `undefined` because it is not supported and should never happen.
if we want to be more strict about it maybe raise an error rather than just log, but I am honestly not sure

---

I thought from the conversation yesterday that it was better to let this pass here and become a merge error, no?
(re error vs log - I wasn't sure either, so if you do want to change it lmk)

---

ok.
we should still log both cases IMO (not just if the length is different).
and maybe change the log severity to error because it really shouldn't happen?